File: src/main/java/com/app/Message_Backend/pojo/Message.java
package com.app.Message_Backend.pojo;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Date;

@Entity(name = "Messages")
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String text;
    private Date timeSent;
    private Long conversationId;
    private Long senderId;

    public Message() {}

    public Message(String text, Long conversationId, Long senderId) {
        this.text = text;
        this.conversationId = conversationId;
        this.senderId = senderId;
    }

    public Message(String text, Long conversationId, Long senderId, Date timeSent) {
        this.text = text;
        this.conversationId = conversationId;
        this.senderId = senderId;
        this.timeSent = timeSent;
    }


    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Long getId() {
        return id;
    }

    public Date getTimeSent() {
        return timeSent;
    }

    public void setTimeSent(Date timeSent) {
        this.timeSent = timeSent;
    }

    public Long getConversationId() {
        return conversationId;
    }

    public void setConversationId(Long conversationId) {
        this.conversationId = conversationId;
    }

    public Long getSenderId() {
        return senderId;
    }

    public void setSenderId(Long senderId) {
        this.senderId = senderId;
    }
}

File: frontend/src/dto/Message.ts
export default class Message {
    private id: number;
    private text: string;
    private conversationId: number;
    private senderId: number;


	constructor(id: number, text: string, conversationId: number, senderId: number) {
        this.id = id;
        this.text = text;
        this.conversationId = conversationId;
        this.senderId = senderId; 
	}


    /**
     * Getter $id
     * @return {number}
     */
	public getId(): number {
		return this.id;
	}

    /**
     * Getter $text
     * @return {string}
     */
	public getText(): string {
		return this.text;
	}

    /**
     * Setter $text
     * @param {string} value
     */
	public setText(value: string) {
		this.text = value;
	}

    /**
     * Getter $conversationId
     * @return {number}
     */
	public getConversationId(): number {
		return this.conversationId;
	}

    /**
     * Setter $conversationId
     * @param {number} value
     */
	public setConversationId(value: number) {
		this.conversationId = value;
	}

    /**
     * Getter $senderId
     * @return {number}
     */
	public getSenderId(): number {
		return this.senderId;
	}

    /**
     * Setter $senderId
     * @param {number} value
     */
	public setSenderId(value: number) {
		this.senderId = value;
	}

}